sympy/matrices/matrices.py

```diff
@@ -27,6 +27,12 @@
 
 def _iszero(x):
     """Returns True if x is zero."""
+    if hasattr(x, "free_symbols") and not x.free_symbols:
+        # if we do not have free symbols, we will call _simplify()
+        # without risks to lose some confines for domain of an expression,
+        # because it will be some number
+        x = _simplify(x)
```

Comment:
In general, I always give a -1 to having simplification calls in our code base. If x is an arbitrary matrix, simplify can take inordinate amounts of computation time to complete. An iszero check should be fast. You probably want to read the conversation around issues and pull requests like https://github.com/sympy/sympy/pull/10280.

If you are going to add this simplify call, then I request that you add a benchmark to https://github.com/sympy/sympy_benchmarks so that we track how slow things start to get.